webpack/components/extensions/Hosts/BulkActions/BulkSystemPurposeModal/BulkSystemPurposeModal.js

```diff
@@ -0,0 +1,334 @@
+      }
+      if (selectedUsage !== NO_CHANGE) {
+        sysPurposeParams.purpose_usage = selectedUsage;
+      }
+      if (selectedServiceLevel !== NO_CHANGE) {
+        sysPurposeParams.service_level = selectedServiceLevel;
+      }
+
+      dispatch(bulkUpdateHostSystemPurpose(
+        sysPurposeParams,
+        handleModalClose,
+        handleModalClose,
+      ));
+      apiCallsMade = true;
+    }
+
+    // Call release version endpoint if release version changed
+    if (selectedReleaseVersion !== NO_CHANGE) {
+      const releaseParams = {
+        ...baseParams,
+        release_version: selectedReleaseVersion,
+      };
+
+      dispatch(bulkUpdateHostReleaseVersion(
+        releaseParams,
+        apiCallsMade ? null : handleModalClose, // Only close modal once
+        handleModalClose,
+      ));
+      apiCallsMade = true;
+    }
+
+    // If we didn't make any API calls, just close the modal
+    if (!apiCallsMade) {
+      handleModalClose();
+    }
+  };
+
+  const modalActions = [
+    <Button
+      key="save"
+      ouiaId="bulk-system-purpose-modal-save-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={!hasChanges() || updateStatus === STATUS.PENDING}
+      isLoading={updateStatus === STATUS.PENDING}
+    >
+      {__('Save')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-system-purpose-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change system purpose')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-system-purpose-modal"
+      key="bulk-system-purpose-modal"
+      ouiaId="bulk-system-purpose-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-system-purpose-description">
+          <FormattedMessage
+            defaultMessage="This will update the system purpose for {hosts}. Not all releases may be compatible with all selected hosts."
+            values={{
+              hosts: (
+                <strong>
+                  <FormattedMessage
+                    defaultMessage="{count, plural, one {# {singular}} other {# {plural}}}"
+                    values={{
+                      count: selectedCount,
+                      singular: __('selected host'),
+                      plural: __('selected hosts'),
+                    }}
+                    id="bulk-system-purpose-count"
+                  />
+                </strong>
+              ),
+            }}
+            id="bulk-system-purpose-description-i18n"
+          />
+        </Text>
+      </TextContent>
+
+      <Form>
+        <FormGroup label={__('Role')} fieldId="bulk-system-purpose-role">
+          <FormSelect
+            value={selectedRole}
+            onChange={event => setSelectedRole(event.target.value)}
```

Comment:
How about `onChange={(_event, value) => setSelectedRole(value)}`?
Same for other places.